### PR TITLE
feat: subs in subscribe response & versioning information

### PR DESCRIPTION
**Pull request by chris13524**
- Adds `sbs` which is the list of subscriptions to `notifySubscribe`, `notifyUpdate`, and `notifyDelete` responses to simplify client-side implementation to avoid listening for an additional subscriptionsChanged. [Context](https://walletconnect.slack.com/archives/C044SKFKELR/p1701870285066439?thread_ts=1701804693.064159&cid=C044SKFKELR).
- Adds `mjv` to all JWTs which enables each side to tell what version the other side is using and if there have been any major breaking changes to the API in-use.
  - Notify Server can use this to tell if a client supports the new `sbs` response field and thus will avoid sending a duplicate subscriptionsChanged request to supporting clients
  - Clients can use this to tell if Notify Server has broken support for an old API version, and for clients to fail more gracefully such as logging "please update your SDK version". For example this could happen if support for an insecure SIWE statement is dropped, or there are other major protocol improvements that need to be made and support dropped after a sufficient majority of clients migrate.
- ~~Adds `sdk` versioning information for analytics purposes~~

### CODE REVIEW

File: docs/specs/clients/notify/authentication.md
Comment (Elyniss) [inline docs/specs/clients/notify/authentication.md:13]: Why is this major version of the API level? Wouldn't it be better to call it schema version?

Comment (chris13524) [reply docs/specs/clients/notify/authentication.md:13]: No specific/strong reason I choose this word, I think schema version would work. Not sure yet if each JWT might have different version #s or if it would be for the entire protocol, depends on how we end up using it. But for now we need to differentiate between `0` and `1` for the `sbs` response

```
mjv - major version of the schema being used as a string, currently `"1"`
```

File: docs/specs/clients/notify/authentication.md
Comment (Elyniss) [inline docs/specs/clients/notify/authentication.md]: Do we want analytics on the protocol level? This data is already in the relay client. Seems like a duplicate

Comment (chris13524) [inline docs/specs/clients/notify/authentication.md]:
```suggestion
```

Comment (chris13524) [reply docs/specs/clients/notify/authentication.md]: Makes sense, removed in https://github.com/WalletConnect/walletconnect-specs/pull/182/files#r1420739990

**Comment by chris13524** [comment]
Added `sbs` to `notifyUpdate` and `notifyDelete` responses too

**Comment by chris13524** [comment]
Technically there is a race condition with this spec where different changes happen in different orders, resulting in an incorrect state in the client. However a restart will quickly resolve this anyway. Also it's missing pagination for large lists of subscriptions. Things to improve for a later revision of the spec.

File: docs/specs/clients/notify/authentication.md
Comment (Elyniss) [reply docs/specs/clients/notify/authentication.md]: Please commit

File: docs/specs/clients/notify/rpc-methods.md
Comment (Elyniss) [inline docs/specs/clients/notify/rpc-methods.md]: <3

**Comment by Elyniss** [comment]
LGTM please ask for re-review once `sdk` is removed

File: docs/specs/clients/notify/rpc-methods.md
Comment (Elyniss) [inline docs/specs/clients/notify/rpc-methods.md]:
```suggestion
Watchers will only be notified of changes from _other_ clients. I.e. if a subscription change is made by `iss`, then as a response to that `iss`'s request the latest subscriptions will be provided. But it will not receive a `wc_notifySubscriptionsChanged` request.

```

Since there are no iss2 I would remove to avoid confusion

Comment (chris13524) [reply docs/specs/clients/notify/rpc-methods.md]: I need to improve the wording here overall

**Review by Elyniss — APPROVED** [review]

**Comment by chris13524** [comment]
Now blocked by client public API changes

**Review by flypaper0 — APPROVED** [review]

**Review by devceline — APPROVED** [review]
LGTM

**Pull request merged**
